geckomat/change_model/makeEcModel.m

```diff
@@ -111,54 +104,28 @@
 %6: Sort reactions, so that reversible and isoenzymic reactions are kept near
 model=sortIdentifiers(model);
 
-%7: Make a reduced S-matrix, with zero entries for metabolites to be
-%ignored for defining Kcat values.
-%TODO: have this loaded by a separate function, so that these lists can be
-%set as input parameters to the current function, avoiding repeated loading
-%of these files if multiple models are built.
-reducedS    = model.S;
-%Remove simple substrates
-fid         = fopen(fullfile(geckoPath,'databases','smallMets.tsv'));
-fileContent = textscan(fid,'%q','Delimiter','\t');
-fclose(fid);
-metsToIgnore = contains(model.metNames,fileContent{1});
-reducedS(metsToIgnore,:) = 0;
-
-%Remove currency metabolites
-fid         = fopen(fullfile(geckoPath,'databases','currencyMets.tsv'));
-fileContent = textscan(fid,'%q %q','Delimiter','\t');
-fclose(fid);
-currMets.sub = fileContent{1};
-currMets.pro = fileContent{2};
-
-%Reaction needs to involve both metabolites from a pair (=row in TSV file)
-for i=1:numel(currMets.sub)
-    idS = contains(model.metNames,currMets.sub{i});
-    idP = contains(model.metNames,currMets.pro{i});
-    [~,hasS] = find(reducedS(idS,:));
-    [~,hasP] = find(reducedS(idP,:));
-    hasBoth  = intersect(hasS,hasP);
-    reducedS([idS,idP],hasBoth) = 0;
+%7: Make ec-extension structure, one for gene-associated reaction.
+rxnWithGene  = find(sum(model.rxnGeneMat,2));
+ec.rxns      = model.rxns(rxnWithGene);
+emptyCell    = cell(numel(rxnWithGene),1);
+emptyVect    = zeros(numel(rxnWithGene),1);
+if isfield(model,'eccodes')
+    ec.eccodes = model.eccodes(rxnWithGene);
+else
+    ec.eccodes = emptyCell;
+    %TODO: loading of external ec-code database, possibly from uniprot, which
+    %should directly be parsed to the model, so that the eccodeDB entries match
+    %model.rxns. This is problematic, because Uniprot can contain multiple
+    %eccodes, while eccodes might not exactly match the reactions that are
+    %catalyzed. Only needed for classic GECKO matching, might just copy how it
+    %was dealt with there
 end
-
-%8: Make ec-extension structure, one for each kcat value
-emptyCell    = cell(numel(model.rxns)*5,1); %Prepare plenty of space
-emptyVect    = zeros(numel(model.rxns)*5,1);
-ec.rxns      = emptyCell;
-ec.eccodes   = emptyCell; % Can also be parsed from model. Only needed for classic GECKO matching
-ec.rxnEnzMat = zeros(numel(model.rxns)*5,numel(model.genes)); % Non-zeros will indicate the number of subunits
-%ec.substrate = emptyCell; 
-%ec.reverse   = logical(emptyVect); % Whether reaction is in reverse. Might be unnecessary, can also be gathered from the reaction ID ('_REV')
+ec.rxnEnzMat = zeros(numel(rxnWithGene),numel(model.genes)); % Non-zeros will indicate the number of subunits
 ec.kcat      = emptyVect;
```

Comment:
Something needs to be done here for GeckoLight - the question is where to apply the max(kcats from isozymes).

Reply:
`makeEcModel` is only making the basic structure, it is not yet populated with kcat values.